common/web/types/test/ldml-keyboard/test-pattern-parser.ts

```diff
@@ -379,6 +379,12 @@ describe('Test of nfd_markers()', () => {
       ["\u0995\uFFFF\u0008\u0001\u09CB", "\u0995\uFFFF\u0008\u0001\u09C7\u09BE"],
       ["\u0995\u09BE\uFFFF\u0008\u0001\u09C7"],
       ["\u03B5\uFFFF\u0008\u0001\u0344", "\u03B5\uFFFF\u0008\u0001\u0308\u0301"],
+
+      // double marker -
+      ["e\uffff\u0008\u0001\u0300\u0320\u0300", "e\u0320\uffff\u0008\u0001\u0300\u0300"],
+      // double marker - no change
+      ["e\u0320\uffff\u0008\u0001\u0300\u0300", "e\u0320\uffff\u0008\u0001\u0300\u0300"],
+
```

Comment:
@jahorton this is your example..

Reply:
So... the marker is `\u0001`?  That's the part before the first entry of the doubled `\u0300`.  (I'd have expected `\uffff`, but that wouldn't make it "my example".)

Assuming so...

```suggestion

      // double marker -
      ["e\uffff\u0008\u0001\u0300\u0320\u0300", "e\u0320\uffff\u0008\u0001\u0300\u0300"],
      // double marker - no change
      ["e\u0320\uffff\u0008\u0001\u0300\u0300", "e\u0320\uffff\u0008\u0001\u0300\u0300"],
      // double marker - in front of second
      ["e\u0320\uffff\u0008\u0300\u0001\u0300", "e\u0320\uffff\u0008\u0300\u0001\u0300"],
```

We want to be sure it's handled right, whichever of the duplicated entries it was originally in front of.

Reply:
Marker is `\uffff\u0008{marker_id}`.

Reply:
Also, for extra robustness... how well does the solution hold up if the same mark gets repeated 3+ times within the same segmentation cluster?  (Or... is that too niche for us to care about here?)

The case I'd most expect to fail would be something like this - with the marker prepended to the _middle_ instance of the triplicated char.
```typescript
// triple diacritic + marker - reordering needed
["e\u0300\uffff\u0008\u0001\u0300\u0320\u0300", "e\u0320\u0300\uffff\u0008\u0001\u0300\u0300"],
```

Reply:
@jahorton it should hold up - it handles 1:1 all input nfd chars.

Reply:
Then it should hold up with the suggested extra unit test cases, right?